doc/adr/0010-reducing-snapshot-size.md

```diff
@@ -0,0 +1,343 @@
+
+One of the advantages of this representation are flexible aggregation schemes.
+
+Simple scheme: 
+* if ordinal is divisible by 1000 do aggregate since 0 (genesis)
+* otherwise do aggregate since last ordinal divisible by 1000
+```mermaid
+graph RL;
+    A(ordinal = 1999, since = 1001)
+    B(ordinal = 1000, since = 0)
+    A-->B;
+```
+
+More complicated scheme:
+* if ordinal is divisible by 1000 do aggregate since 0 (genesis)
+* if ordinal is divisible by 100 do aggregate since last ordinal divisible by 1000
+* if ordinal is divisible by 10 do aggregate since last ordinal divisible by 100
+* otherwise, do aggregate since last ordinal divisible by 10
+```mermaid
+graph RL;
+    A(ordinal = 1999, since = 1991)
+    B(ordinal = 1990, since = 1901)
+    C(ordinal = 1900, since = 1001)
+    D(ordinal = 1000, since = 0)
+    A-->B;
+    B-->C;
+    C-->D;
+```
+
+### Aggregated view resolution
+
+Algorithm for resolving an aggregated view of all data - from the genesis to the latest snapshot:
+1. Variable initialization:
+    * `snapshot := <latest-snapshot>`
+    * `acc := snapshot.info.aggregated`
+2. Loop:
+    * if `snapshot.info.aggregated.since` is equal to `0` return `acc`
+    * otherwise, get the snapshot with ordinal `snapshot.info.aggregated.since` and assign it to `snapshot`
+    * combine `acc` with `snapshot.info.aggregated` and assign it to `acc`
+
+
+### Example of first 2000 snapshots
+
+This example uses a **Simple scheme** presented in section [Aggregation schemes](#aggregation-schemes). 
+
+Simplifications:
+* all snapshots that are not listed below didn't contain any transactions
+* rewards were 0 for all snapshots
+* some fields were omitted for clarity
+
+Snapshots:
+- [Snapshot 0 (genesis)](#snapshot-0)
+- [Snapshot 1](#snapshot-1)
+- [Snapshot 999](#snapshot-999)
+- [Snapshot 1000](#snapshot-1000)
+- [Snapshot 1001](#snapshot-1001)
+- [Snapshot 1999](#snapshot-1999)
+- [Snapshot 2000](#snapshot-2000)
+
+#### Snapshot 0
+
+```json
+{
+  "ordinal": 0,
+  "blocks": [],
+  "info": {
+    "partial": {
+      "balances": {
+        "A": 20,
+        "B": 20
+      }
+    },
+    "aggregated": {
+      "since": 0,
+      "balances": {
+        "A": 20,
+        "B": 20
+      }
+    }
+  }
+}
+```
+
+#### Snapshot 1
+
+```json
+{
+  "ordinal": 1,
+  "blocks": [
+    {
+      "transactions": [
+        {
+          "source": "B",
+          "destination": "C",
+          "amount": 10
+        }
+      ]
+    }
+  ],
+  "info": {
+    "partial": {
+      "balances": {
+        "B": 10,
+        "C": 10
+      }
+    },
+    "aggregated": {
+      "since": 1,
+      "balances": {
+        "B": 10,
+        "C": 10
+      }
+    }
+  }
+}
+```
+
+#### Snapshot 999
+
+```json
+{
+  "ordinal": 999,
+  "blocks": [
+    {
+      "transactions": [
+        {
+          "source": "C",
+          "destination": "D",
+          "amount": 10
+        }
+      ]
+    }
+  ],
+  "info": {
+    "partial": {
+      "balances": {
+        "B": 10,
```

Comment:
B wasn't influenced by any transaction so I guess it shouldn't be in `partial`

Reply:
Yep, thanks for catching this!